### PR TITLE
Small refactorings in the BKGServiceImpl

### PR DESCRIPTION
**Pull request by pimg**

### CODE REVIEW

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java [hunk 1]
```diff
@@ -646,15 +638,10 @@ private Mono<Tuple2<TransportEvent, TransportEvent>> fetchTransportEventByTransp
...
-        .flatMap(
-            x -> {
-              if (x.getVesselID() == null) {
-                return Mono.empty();
-              }
-              return vesselRepository.findById(x.getVesselID());
-            });
+    return fetchTransportCallById(transportCallId)
+        .filter(transportCall -> Objects.nonNull(transportCall.getVesselID()))
+        .switchIfEmpty(Mono.empty())
```
Comment (nt-as) [inline src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java:643]: `         .switchIfEmpty(Mono.empty())` is a no op. Did you mean to add something else here (like a Mono.error)?

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java [hunk 2]
```diff
@@ -505,11 +511,15 @@ public Mono<ShipmentTO> getShipmentByCarrierBookingReference(
                           .doOnNext(shipmentTO::setShipmentCutOffTimes),
                       fetchShipmentLocationsByBookingID(shipment.getBookingID())
                           .doOnNext(shipmentTO::setShipmentLocations),
-                      fetchCarrierClausesByShipmentID(shipment.getShipmentID())
+                      carrierClauseService
+                          .fetchCarrierClausesByShipmentID(shipment.getShipmentID())
```
Comment (mp-assco) [inline src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java:515]: Local function has a null check, but the service implementation doesn't. Is this going to be a problem?

Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java:515]: I don't believe so, shipmentID is the primary key column of the shipmentTable so the db constraints will ensure no shipments exist without an ID

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java (hunk 1 above)
Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java]: No, took a too literal way of translating the if statement to a filter. The switchIfEmpty is removed

**Review by nt-as — APPROVED** [review]

**Review by douradofrazer — APPROVED** [review]

**Pull request merged**